dnf/package.py

```diff
@@ -225,9 +225,9 @@ def schemes_filter(url_list):
                 if schemes:
                     s = dnf.pycomp.urlparse.urlparse(url)[0]
                     if s in schemes:
-                        return url + self.location
+                        return os.path.join(url, self.location)
```

Comment:
@j-mracek considering DNF doesn't run on Windows (os.path.join() generates backslashes there), the code should do the job.
Although os.path.join() can be quite fragile.
If self.location starts with '/', then the join discards the whole url part.

urljoin() is also tricky, it expects trailing '/' otherwise the last element is treated as a file and removed from url.

Couldn't you simply revert this piece of patch and stick with the original '+'?
It's no change to the original and even if it's not nice, it's bullet proof.
We can always come up with a better patch later.